File: app/src/main/java/com/scop/org/minesweeper/LoadingActivity.java
package com.scop.org.minesweeper;

import android.app.Activity;
import android.content.res.Configuration;
import android.os.Bundle;
import android.support.v7.app.AppCompatActivity;
import android.view.Window;
import android.view.WindowManager;

import com.scop.org.minesweeper.control.ScreenProperties;

/**
 * An example full-screen activity that shows and hides the system UI (i.e.
 * status bar and navigation/system bar) with user interaction.
 */
public class LoadingActivity extends Activity {

	private static Activity activeActivity;

	@Override
	protected void onCreate(Bundle savedInstanceState) {
		super.onCreate(savedInstanceState);

		requestWindowFeature(Window.FEATURE_NO_TITLE);
		getWindow().setFlags(WindowManager.LayoutParams.FLAG_FULLSCREEN, WindowManager.LayoutParams.FLAG_FULLSCREEN);

		setContentView(R.layout.activity_loading);
		activeActivity = this;
	}

	public static Activity getActiveActivity(){
		return activeActivity;
	}

	@Override
	protected void onDestroy() {
		super.onDestroy();
		activeActivity = null;
	}
}

File: app/src/main/java/org/oar/minesweeper/LoadingActivity.kt
package org.oar.minesweeper

import android.app.Activity
import android.content.Intent
import android.os.Bundle
import android.view.Window
import android.view.WindowInsets
import org.oar.minesweeper.models.Grid
import org.oar.minesweeper.utils.ActivityUtils.animateStartActivity
import org.oar.minesweeper.utils.GridUtils.generate
import kotlin.reflect.full.createInstance

/**
 * An example full-screen activity that shows and hides the system UI (i.e.
 * status bar and navigation/system bar) with user interaction.
 */
class LoadingActivity : Activity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        requestWindowFeature(Window.FEATURE_NO_TITLE)
        setContentView(R.layout.activity_loading)
        window.insetsController?.hide(WindowInsets.Type.statusBars())

        val grid = this.intent.extras?.getSerializable("grid") as Grid? ?:
            throw NullPointerException("A grid was expected")

        grid.generate {
            val intent = Intent(this, GameActivity::class.java)
            intent.putExtra("grid", grid)
            intent.putExtra("options", it)
            animateStartActivity(intent, true)
        }
    }
}
